meta-cli/src/cli/typegate.rs

```diff
@@ -22,33 +24,23 @@ impl Action for Typegate {
     }
 }
 
-pub fn command(cmd: Typegate, gen_args: GenArgs) -> Result<()> {
-    let cwd = match cmd.root_dir {
-        Some(path) => path.clone().normalize()?.into_path_buf(),
-        None => gen_args.dir()?,
-    };
-    let main_module = cwd.join("typegate/src/main.ts");
-    let permissions = mt_deno::deno::deno_runtime::permissions::PermissionsOptions {
-        allow_run: Some(["hostname"].into_iter().map(str::to_owned).collect()),
-        allow_sys: Some(vec![]),
-        allow_env: Some(vec![]),
-        allow_hrtime: true,
-        allow_write: Some(
-            ["tmp"]
-                .into_iter()
-                .map(std::str::FromStr::from_str)
-                .collect::<Result<_, _>>()?,
-        ),
-        allow_ffi: Some(vec![]),
-        allow_read: Some(
-            ["."]
-                .into_iter()
-                .map(std::str::FromStr::from_str)
-                .collect::<Result<_, _>>()?,
-        ),
-        allow_net: Some(vec![]),
-        ..Default::default()
-    };
-    mt_deno::run_sync(main_module, permissions, std::sync::Arc::new(Vec::new));
+pub fn command(cmd: Typegate, _gen_args: GenArgs) -> Result<()> {
+    let runtime = typegate_core::runtime();
+    const BASE_URL: &str =
```

Comment:
A thought on the first item:

A consideration that went into choosing `xtask` as the one to use for typegate dev was that adding the deno dependency to meta-cli by default almost doubles the link time. Meta-cli is used in the test suite and gets frequently rebuilt, more so than xtask.

Reply:
Thoughts on the second point:

If the intention is to avoid touching the network, we ought to vendor and embed the ecma dependencies to the binary as well. 

On the question of choosing which version to link to on GitHub (with r current approach), I was thinking of using a build time env var to set which commit it points to. If not being built in a git repo, it fallbacks to the tagged commit that corresponds to `CARGO_CRATE_VERSION`.

Reply:
I meant to keep the xtask also here, so that in dev we use base_url to point on the local filesystem. Do I miss something here?

Can you explain why we need ecma if we include/vendor the typescript files? I thought we may only need to decompress the bundle files and change base_url to them.

Reply:
If `xtask typegate` is the default typegate runner at dev time, there's no need to add a dev specific typegate command to `meta-cli`. The `typegate` command on `meta-cli` is also overridable using flags if you ever need to run from the local source files but that's a very rare scenario.

And regarding vendoring, I meant to say that if the motivation to use the embedded flate approach is to avoid loading the typegate source files from the network...unless we also embed the third-party dependencies of typegate like `sentry` and `zod`, we end up loading files from the network either ways.

Reply:
Make sense! For sentry/zod/other dependencies, they could be cached by [that](https://github.com/metatypedev/metatype/blob/main/dev/Dockerfile#L63) but I agree the current approach is more flexible forgetting about the network. Using the commit is perfect, no need to have fallback :+1:!